File: csharp/DesiArchitect.DesignPatterns/06_Adapter/After/DelhiveryClient.cs
namespace DesiArchitect.DesignPatterns.Adapter.After;

// Simulated Delhivery SDK - completely different method names and shapes.
// Same point as Bluedart: the vendor's payload/response types belong with the vendor SDK.
public class DelhiveryClient
{
    public DelhiveryTrackingResponse DispatchShipment(DelhiveryShipmentPayload payload)
    {
        Console.WriteLine($"     [Delhivery SDK] DispatchShipment -> Waybill:{payload.WaybillRef}, Drop:{payload.DropAddress}");
        return new DelhiveryTrackingResponse($"DL-{payload.WaybillRef}-TRK");
    }
}

public record DelhiveryShipmentPayload(string WaybillRef, string DropAddress, decimal PackageWeight);
public record DelhiveryTrackingResponse(string TrackingNumber);

File: typescript/src/06-adapter/after/delhivery-client.ts
export interface DelhiveryShipmentPayload {
  waybillRef: string;
  dropAddress: string;
  packageWeight: number;
}

export interface DelhiveryTrackingResponse {
  trackingNumber: string;
}

export class DelhiveryClient {
  dispatchShipment(payload: DelhiveryShipmentPayload): DelhiveryTrackingResponse {
    console.log(`     [Delhivery SDK] DispatchShipment -> Waybill:${payload.waybillRef}, Drop:${payload.dropAddress}`);
    return { trackingNumber: `DL-${payload.waybillRef}-TRK` };
  }
}